rdt/hyper_transformer.py

```diff
@@ -212,6 +212,44 @@ def update_default_data_type_transformers(self, new_data_type_transformers):
         self.default_data_type_transformers.update(new_data_type_transformers)
         self._unfit()
 
+    def update_transformers(self, column_name_transformer):
+        """Update any of the transformers assigned to each of the column names.
+
+        Args:
+            column_name_transformer(dict):
+                Dict mapping column names to transformers to be used for that column.
+        """
+        if self._fitted:
+            warning_msg = (
+                "Warning: For this change to take effect, please refit your data using 'fit' "
+                "or 'fit_transform'."
+            )
+            print(warning_msg)  # noqa: T001
+
+        if len(self.field_transformers) == 0:
+            tip_msg = (
+                'Tip: Use the `detect_initial_config` method to pre-populate all the sdtypes '
+                'and transformers from your dataset.'
+            )
+            print(tip_msg)  # noqa: T001
+
+        for column_name, transformer in column_name_transformer.items():
+            current_transformer = self.field_transformers.get(column_name)
+            if current_transformer:
+                current_transformer_type = current_transformer.get_input_type()
```

Comment:
Instead of checking the current transformer I think we should check the `field_sdtypes` dict and compare to what is set there